File: Xamarin/Framework/Datas/EventDetail.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Framework.Datas
{
    /// <summary>
    /// the class contains all the information for an event
    /// </summary>
    public class EventDetail
    {
        public string name { get; set; }
        public DateTime date { get; set; }
        public int available_seats { get; set; }
        public float price { get; set; }
        public string venue { get; set; }
        public List<string> labels { set; get; }
    }
}

File: Android/app/src/main/java/com/framework/datas/EventDetail.java
package com.framework.datas;

import java.util.Date;
import java.util.*;

/// the event detail class
public class EventDetail
{
    public String name;
    public Date date;
    public int available_seats;
    public float price;
    public String venue;
    public List<String> labels;

    /// determine whether this event has seats
    public boolean HasSeats () {
        return available_seats > 0;
    }

    /// determine whether this event has all the targeted labels
    public boolean HasLabels(List<String> target_labels) {
        return labels.containsAll(target_labels);
    }
}
